users/renderers.py

```diff
@@ -0,0 +1,23 @@
+import json
+from typing import Any, Optional, Dict
+
+from rest_framework.renderers import JSONRenderer
+
+
+class UserJSONRenderer(JSONRenderer):
+    charset = 'utf-8'
+
+    def render(self, data: dict, media_type:str='', renderer_context:Optional[Dict[str, Any]]=None) -> Any:
+        errors = data.get('errors', None)
+
+        token = data.get('token', None)
+
+        if errors is not None:
+            return super(UserJSONRenderer, self).render(data)
+
+        if token is not None and isinstance(token, bytes):
+            data['token'] = token.decode('utf-8')
+
+        return json.dumps({
+            'user': data
+        })
```

Comment:
```suggestion
    def render(self, data: dict, media_type:str='', renderer_context:Optional[Dict[str, Any]]=None) -> Any:
        if data.get('errors', None):
            return super().render(data)

        if (token := data.get('token', None)) and isinstance(token, bytes):
            data['token'] = token.decode('utf-8')

        return json.dumps({
            'user': data
        })
```
1) Избавляемся от лишней переменной errors, которая будет использована 1 раз
2) Нет смысла проверять на is not None, если там будет пустой список, пустая строка, либо же непосредственно None - то это = False и значит, что мы не зайдем в такой if
3) 13 строчка переезжает по своей логике в if условие. Операция `:=` - это операция присвоения или морж (обычно так именуют её, хз почему именно такое название), позволяет "на месте" задать значение какой-то переменной, очень удобная конструкция, почитай про нее и про юз-кейсы, не раз еще встретишь в работе места, где это удобно
4) опять заменил is not None на проверку объекта `if token`

Reply:
А, еще забыл подметить, на это должен даже Pycharm ругаться, это вроде по PEP несоответствие требованиям к коду. Нет давно уже надобности в super указывать self/объект текущего класса. Это нужно в исключительных ситуациях, я еще ни разу не сталкивался с тем, что мне бы понадобилось самому прописывать класс